### ISSUE TITLE
AsgCapacityProvider: Cannot disable drainhooks after taskDrainTime got deprecated

### ISSUE THREAD

**Issue by tommydongaws** [aws/aws-cdk#24641]
### Describe the bug

Previously in CDKv1 [1] there was a taskDrainTime property which could be used to disable drainhooks. After it got removed, there is no way to remove the drainhooks without enabling "enableManagedTerminationProtection".

From the code here:
` if (!options.taskDrainTime || options.taskDrainTime.toSeconds() !== 0) {
      new InstanceDrainHook(autoScalingGroup, 'DrainECSHook', {
        autoScalingGroup,
        cluster: this,
        drainTime: options.taskDrainTime,
        topicEncryptionKey: options.topicEncryptionKey,
      });
    }`

It look like v2 is still using taskDrainTime to decide whether or not to disable the drain hook which doesn't seem right. Could you add taskDrainTime or a diff option to disable the drainhook?

[1] https://docs.aws.amazon.com/cdk/api/v1/docs/@aws-cdk_aws-ecs.AsgCapacityProviderProps.html
[2] https://github.com/aws/aws-cdk/blob/main/packages/@aws-cdk/aws-ecs/lib/cluster.ts#L559

### Expected Behavior

For there to be no drainhook or an option to disable the drainhook.

### Current Behavior

The code looks for the deprecated property "taskDrainTime"  to decide make a drainhook or not. The only way to disable it is to enable enableManagedTerminationProtection which cause the value of taskDrainTime to be 0.

### Reproduction Steps

When you run this it will create a drain hook.

import * as cdk from 'aws-cdk-lib';
import { Construct } from 'constructs';
import * as ec2 from 'aws-cdk-lib/aws-ec2';
import * as autoscaling from 'aws-cdk-lib/aws-autoscaling';
import * as ecs from 'aws-cdk-lib/aws-ecs';

export class CdktestStack extends cdk.Stack {
  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    // VPC
    const vpc = new ec2.Vpc(this, 'TheVPC', {
      ipAddresses: ec2.IpAddresses.cidr('10.0.0.0/16'),
    })

     const cluster = new ecs.Cluster(this, 'Cluster', {
      vpc,
    });
    
    const autoScalingGroup = new autoscaling.AutoScalingGroup(this, 'ASG', {
      vpc,
      instanceType: new ec2.InstanceType('t2.micro'),
      machineImage: ecs.EcsOptimizedImage.amazonLinux2(),
      minCapacity: 0,
      maxCapacity: 100,
    });
    
    const capacityProvider = new ecs.AsgCapacityProvider(this, 'AsgCapacityProvider', {
      autoScalingGroup: autoScalingGroup,
      enableManagedTerminationProtection: false,
      enableManagedScaling: false,
      
    });
    
    cluster.addAsgCapacityProvider(capacityProvider,{});
    
    const taskDefinition = new ecs.Ec2TaskDefinition(this, 'TaskDef');
    
    taskDefinition.addContainer('web', {
      image: ecs.ContainerImage.fromRegistry('amazon/amazon-ecs-sample'),
      memoryReservationMiB: 256,
    });
    
    new ecs.Ec2Service(this, 'EC2Service', {
      cluster,
      taskDefinition,
      capacityProviderStrategies: [
        {
          capacityProvider: capacityProvider.capacityProviderName,
          weight: 1,
        },
      ],
    });
  }
}

### Possible Solution

Add an option to disable the drainhook.

### CDK CLI Version

2.69.0 (build 60a5b2a)

### Node.js Version

v18.15.0

### OS

Windows 11

### Language

Typescript

**Comment by pahud** [comment]
Hi

Yes according to the doc, [taskDrainTime](https://docs.aws.amazon.com/cdk/api/v1/docs/@aws-cdk_aws-ecs.AsgCapacityProviderProps.html#taskdraintimespan-classapi-icon-api-icon-deprecated-titlethis-api-element-is-deprecated-its-use-is-not-recommended%EF%B8%8Fspan) has been deprecated and should enable  [enableManagedTerminationProtection](https://docs.aws.amazon.com/cdk/api/v1/docs/@aws-cdk_aws-ecs.AsgCapacityProviderProps.html#enablemanagedterminationprotection) instead. Are you trying to disable drain hooks with managedTerminationProtection disabled? Can you elaborate more on your use case?

**Comment by tommydongaws** [comment]
Hi,
Yes, Im trying to disable drain hooks with managedTerminationProtection disabled. Then attach a custom drain hook.

**Comment by pahud** [comment]
Yeah I am afraid you will need to enable managedTerminationProtection now. Is there any reason you can't enable it?

**Comment by pahud** [comment]
Reopening this issue per request. If there's any concern please add the comment below to avoid closing for staleness. Thank you.

**Comment by mrlikl** [comment]
The issue still exists with the latest CDK version. If a cx wishes to disable all the hooks, its not possible. The use-case is even if we flip `enableManagedTerminationProtection` and `enableManagedDraining` to [false](https://github.com/aws/aws-cdk/blob/8318e7968c441ad565139e5faa72977a95099cd2/packages/aws-cdk-lib/aws-ecs/lib/cluster.ts#L455)

```
 taskDrainTime: (provider.enableManagedTerminationProtection || provider.enableManagedDraining)? Duration.seconds(0) : options.taskDrainTime,
```

the resultant template would have hooks in it. 

Because [here](https://github.com/aws/aws-cdk/blob/8318e7968c441ad565139e5faa72977a95099cd2/packages/aws-cdk-lib/aws-ecs/lib/cluster.ts#L562), since taskDrainTime is not exposed, its becoming undefined and the if is always true.
```
 if (!options.taskDrainTime || options.taskDrainTime.toSeconds() !== 0) {
      new InstanceDrainHook(autoScalingGroup, 'DrainECSHook', {
        autoScalingGroup,
        cluster: this,
        drainTime: options.taskDrainTime,
        topicEncryptionKey: options.topicEncryptionKey,
      });
    }
```

Maybe needs to be changed to ?

```
 if (options.taskDrainTime && options.taskDrainTime.toSeconds() !== 0) 
```

**Comment by pahud** [comment]
Sounds like a bug we should fix.

At this moment, you probably would want to remove them using `tryRemoveChild`. Check my full code below:

```ts
export class DummyStack2 extends Stack {
  constructor(scope: Construct, id: string, props: StackProps) {
    super(scope, id, props);

    const vpc = getDefaultVpc(this);
    const cluster = new ecs.Cluster(this, 'Cluster', {
      vpc,
    });
    
    const autoScalingGroup = new autoscaling.AutoScalingGroup(this, 'ASG', {
      vpc,
      instanceType: new ec2.InstanceType('t2.micro'),
      machineImage: ecs.EcsOptimizedImage.amazonLinux2(),
      minCapacity: 0,
      maxCapacity: 100,
    });
    
    const capacityProvider = new ecs.AsgCapacityProvider(this, 'AsgCapacityProvider', {
      autoScalingGroup: autoScalingGroup,
      enableManagedTerminationProtection: false,
      enableManagedScaling: false,
      
    });
    
    cluster.addAsgCapacityProvider(capacityProvider,{});
    
    const taskDefinition = new ecs.Ec2TaskDefinition(this, 'TaskDef');
    
    taskDefinition.addContainer('web', {
      image: ecs.ContainerImage.fromRegistry('amazon/amazon-ecs-sample'),
      memoryReservationMiB: 256,
    });
    
    new ecs.Ec2Service(this, 'EC2Service', {
      cluster,
      taskDefinition,
      capacityProviderStrategies: [
        {
          capacityProvider: capacityProvider.capacityProviderName,
          weight: 1,
        },
      ],
    });

    autoScalingGroup.node.tryRemoveChild('LifecycleHookDrainHook')
    autoScalingGroup.node.tryRemoveChild('DrainECSHook')

  }
}
```

Making it a p2 as there's a workaround but it seems be a low-hanging fruit.

We welcome PRs to get it fixed.
